src/error.rs

```diff
@@ -0,0 +1,82 @@
+use goblin::error::Error as GoblinError;
+use pdb_crate::Error as PDBError;
+use serde::Serialize;
+use std::fmt::{self};
+
+pub type Result<T> = std::result::Result<T, GetSymbolsError>;
+
+#[derive(Debug)]
+pub enum GetSymbolsError {
+    UnmatchedBreakpadId(String, String),
+    PDBError(String),
+    IOError,
+    InvalidInputError(&'static str),
+    GoblinError(String),
```

Comment:
In a previous patch, you were wrapping the original errors from the various crates and not converting them to strings. The change to strings was necessary because you were trying to Serialize this error type directly. But now you have a separate error type for serialization (which I think is good), so you don't have to stringify early any more. Could you go back to wrapping the original error types? I think that would be a bit cleaner. Thanks!

Reply:
If we don't stringify early, we would be converting `PDBError` to a generic `GetSymbolsError:: PDBError` with loss of the `PDBError` message. In that case, we can't pinpoint what PDB message has been fired?

Reply:
You can still get the error message out of the stored PDBError object later, can't you? Instead of calling `pdb_error.to_string()` in `GetSymbolsError::from<PDBError>`, you can call it later in `GetSymbolsError::fmt`.
In practice it won't really make a difference if this code is only ever used through the one WebAssembly entry function. But we may want to use the wasm-independent part of the code for other things in the future.